src/hotspot/share/gc/shenandoah/heuristics/shenandoahOldHeuristics.cpp

```diff
@@ -192,7 +173,69 @@ bool ShenandoahOldHeuristics::prime_collection_set(ShenandoahCollectionSet* coll
     collected_old_bytes += r->garbage();
     consume_old_collection_candidate();
   }
+  return true;
+}
+
+void ShenandoahOldHeuristics::initialize_piggyback_evacs(ShenandoahCollectionSet* collection_set,
+                                                         size_t &evacuated_old_bytes, size_t &collected_old_bytes,
+                                                         uint &included_old_regions, size_t &old_evacuation_reserve,
+                                                         size_t &old_evacuation_budget,
+                                                         size_t &unfragmented_available,
+                                                         size_t &fragmented_available,
+                                                         size_t &excess_fragmented_available) {
```

Comment:
Should all of these reference parameters be members? The API feels complicated now with these three methods passing the same parameters to each other:

```
initialize_piggyback_evacs(...)

prime_collection_set(...)

finalize_piggyback_evacs(...)
```

Could everything just happen within `prime_collection_set` without so many changes to exiting callers?

Reply:
Thanks.  Good suggestion.  Am pursuing this.  Looks much cleaner.